tests/utils/test_atom_array_utils.py

```
@@ -708,3 +710,232 @@ def test_6b8x_indices_roundtrip(self, structure_6b8x):
+
+        assert list(cast(np.ndarray, result[0].chain_id)) == ["A", "B"]
+        assert list(cast(np.ndarray, result[0].res_id)) == [1, 3]
+        assert list(cast(np.ndarray, result[0].atom_name)) == ["CA", "CG"]
+        assert list(cast(np.ndarray, result[0].res_name)) == ["ALA", "LEU"]
+
+    def test_empty_atom_array(self):
+        """Test with empty AtomArray."""
+        atom_array = AtomArray(0)
+        atom_array.coord = np.empty((0, 3))
+
+        with pytest.raises(ValueError, match="Cannot remove atoms from empty AtomArray|Stack"):
```

Ruff (0.15.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)